ldms/src/sampler/spank/slurm_notifier.c

```diff
@@ -663,35 +775,45 @@ jbuf_t make_task_exit_data(spank_t sh, const char *event, const char *context)
 	return jb;
 }
 
-static int nnodes(spank_t sh)
-{
-	uint32_t nnodes;
-	spank_err_t err = spank_get_item(sh, S_JOB_NNODES, &nnodes);
-	if (err)
-		return 0;
-	return nnodes;
-}
-
 int slurm_spank_init(spank_t sh, int argc, char *argv[])
 {
 	spank_context_t context = spank_context();
-	const char *context_str;
-	jbuf_t jb;
-
-	if (0 == nnodes(sh))
-		/* Ignore events before node assignment */
-		return ESPANK_SUCCESS;
+	jbuf_t jb = NULL;
 
 	switch (context) {
+	case S_CTX_ALLOCATOR:
+		/* Called from sbatch/salloc which is likely running on the head node. */
+		jb = make_job_init_data(sh, "init");
+		break;
 	case S_CTX_REMOTE:
-		context_str = "remote";
+		/* Called from slurmstepd running on the node executing the job step */
+		jb = make_step_init_data(sh, "step_init");
 		break;
-	case S_CTX_LOCAL:
 	default:
-		return ESPANK_SUCCESS;
+		break;
 	}
+	if (jb) {
+		DEBUG2("%s", jb->buf);
+		send_event(argc, argv, jb);
+		jbuf_free(jb);
+	}
+	return ESPANK_SUCCESS;
+}
 
-	jb = make_init_data(sh, "init", context_str);
+int slurm_spank_slurmd_init(spank_t sh, int argc, char *argv[])
```

Comment:
This hook, slurm_spank_slurmd_init, needs to be removed entirely.

Reply:
agreed, I had it in there to confirm it wasn't being called